src/store/GlobalState.js

```diff
@@ -0,0 +1,155 @@
+        dispatch({ type: 'RESTORE_EXPENSES', payload: expenses });
+      });
+    } catch (err) {
+      console.log(err);
+      alert('Error loading expenses. Please try again later.');
+    }
+  }, []);
+
+  // Filter for expenses that belong in the selected month
+  useEffect(() => {
+    const filteredExpenses = expenses
+      .filter(expense => {
+        const expenseMonth = getUKFormattedDate(new Date(expense.date), { year: 'numeric', month: 'long' });
+        return selectedMonth === expenseMonth;
+      })
+      .sort((expenseA, expenseB) => new Date(expenseB.date) - new Date(expenseA.date));
+
+    dispatch({ type: 'UPDATE_SELECTED_MONTH_EXPENSES', payload: filteredExpenses });
+  }, [expenses, selectedMonth]);
+
+  useEffect(() => {
+    if (selectedMonthExpenses.length === 0) {
+      changeMonthView(currentMonth);
+    }
+  }, [selectedMonthExpenses, currentMonth]);
+
+  const globalFunctions = {
+    addExpense,
+    editExpense,
+    deleteExpense,
+    changeMonthView,
+    openConfirmDeleteModal,
+    closeConfirmDeleteModal,
+    openEditExpenseModal,
+    closeEditExpenseModal,
+    closeConfirmMonthModal,
+    closeAlert,
+  };
+
+  async function addExpense(expense) {
+    try {
+      const expensesRef = collection(db, 'expenses');
+      await addDoc(expensesRef, expense);
+      dispatch({ type: 'ADD_EXPENSE_SUCCESS', payload: expense });
+    } catch (err) {
+      console.log(err);
+      dispatch({ type: 'ADD_EXPENSE_FAIL' });
+    }
+  }
+
+  async function editExpense(id, newExpenseData) {
+    let didErrorOccur = false;
+
+    try {
+      const expenseRef = doc(db, 'expenses', id);
+      await updateDoc(expenseRef, newExpenseData);
+    } catch (err) {
+      console.log(err);
+      didErrorOccur = true;
+    }
+
+    dispatch({ type: 'EDIT_EXPENSE', payload: didErrorOccur });
+  }
+
+  async function deleteExpense(id) {
+    let didErrorOccur = false;
+
+    try {
+      const expenseRef = doc(db, 'expenses', id);
+      await deleteDoc(expenseRef);
+    } catch (err) {
+      console.log(err);
+      didErrorOccur = true;
+    }
+
+    dispatch({ type: 'DELETE_EXPENSE', payload: didErrorOccur });
+  }
+
+  function changeMonthView(month) {
+    dispatch({ type: 'CHANGE_MONTH_VIEW', payload: month });
+  }
+
+  function openConfirmDeleteModal(expense) {
+    dispatch({ type: 'OPEN_CONFIRM_DELETE_MODAL', payload: expense });
+  }
+
+  function closeConfirmDeleteModal() {
+    dispatch({ type: 'CLOSE_CONFIRM_DELETE_MODAL' });
+  }
+
+  function openEditExpenseModal(expense) {
+    dispatch({ type: 'OPEN_EDIT_EXPENSE_MODAL', payload: expense });
+  }
+
+  function closeEditExpenseModal() {
+    dispatch({ type: 'CLOSE_EDIT_EXPENSE_MODAL' });
+  }
+
+  function closeConfirmMonthModal() {
+    dispatch({ type: 'CLOSE_CONFIRM_MONTH_MODAL' });
+  }
+
+  function closeAlert() {
+    dispatch({ type: 'CLOSE_ALERT' });
+  }
+
+  return (
+    <GlobalContext.Provider value={{ globalState, globalFunctions }}>
```

Comment:
i think i now understand why you are introducing the `globalFunctions` constant. i am not sure however if yet another variable with `global` improves readability. i would like to kick off a small discussion how we can simplify this whole lot. react/reducer introduces a lot of indirection and duplication (i.e. call a function, this function delegates to dispatch using a made-up string, the reducer looks up the made-up string and actually does the work). is there any way to simplify this?